tools/screenshot-test/src/app/firebase.service.ts

```diff
@@ -0,0 +1,198 @@
+import {Injectable} from '@angular/core';
+import * as firebase from 'firebase';
+
+const request = require('request');
+const config = require('../config.json');
+
+
+export class ScreenshotResult {
+  /** PR information: the pull request number */
+  prNumber: string;
+  /** PR information: the sha of the pull request commit */
+  sha: string;
+  /** PR information: The travis job ID */
+  travis: string;
+
+  /** Test result, the test names */
+  testnames: string[];
+  /** Test result: passed or failed. The value can be true if test failed but PR approved by user */
+  result: boolean;
+  /** Test results: passed or failed for each test */
+  results: Map<string, boolean> = new Map<string, boolean>();
+  /**
+   * Test approved: whether the test images are copied to goldens.
+   * The value is the commit SHA and date/time of approval.
+   */
+  approved: string;
+
+  githubStatus: any;
+
+  /** Viewing mode, can be flip, diff, side */
+  mode: 'diff' | 'side' | 'flip' = 'diff';
+  /** Viewing flipping, whether the image is test image or golden image */
+  flipping: boolean = false;
+  /** Viewing collapsed: whether the result should be collapsed/expanded */
+  collapse: boolean[];
+
+  setCollapse(value: boolean) {
+    if (this.collapse) {
+      for (let i = 0; i < this.collapse.length; i++) {
+        this.collapse[i] = value;
+      }
+    }
+  }
+
+  get prLink() {
+    return `https://github.com/${config.repoSlug}/pull/${this.prNumber}`;
+  }
+
+  get commitLink() {
+    return `https://github.com/${config.repoSlug}/commit/${this.sha}`;
+  }
+
+  get travisLink() {
+    return `https://travis-ci.org/${config.repoSlug}/jobs/${this.travis}`;
+  }
+}
+
+
+@Injectable()
+export class FirebaseService {
```

Comment:
This is actually using Firebase directly, instead of going via AngularFire.

@tinayuangao, perhaps consider using AngularFire instead?